bench/workgen/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+project(workgen CXX)
+
+# UseSWIG: Policy option to use the standard target name over a generated name.
+if(POLICY CMP0078)
+    cmake_policy(SET CMP0078 OLD)
+endif()
+
+# UseSWIG: Policy option to honor SWIG_MODULE_NAME.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
+
+find_package(SWIG 3.0 COMPONENTS python REQUIRED)
+include(${SWIG_USE_FILE})
+include(${CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+
+# Determine which wiredtiger target we should use to link against the
+# Python Workgen API. For linking purposes, we need to ensure the wiredtiger
+# library was compiled as position independent code (fPIC). This being achieved
+# either using the 'WITH_PIC' configuration option or building a shared library
+# version of WiredTiger ('ENABLE_SHARED'). Default to whichever option is available
+# (with a preference for a PIC static build), otherwise throw an error if neither
+# build is available.
+set(wiredtiger_target)
+if(ENABLE_STATIC AND WITH_PIC)
+    set(wiredtiger_target wiredtiger_static)
+elseif(ENABLE_SHARED)
+    set(wiredtiger_target wiredtiger_shared)
+else()
+    message(STATIC "Skipping workgen compiliation: Requires either a shared library (ENABLE_SHARED)"
+                    "or a PIC-enabled (WITH_PIC) build of wiredtiger")
+endif()
+
+set(python_libs)
+set(python_version)
+set(python_executable)
+# Source the users Python3 install, extracting the Python 3 version, library and interpreter install paths.
+source_python3_package(python_libs python_version python_executable)
+
+# Compile the intermediate C++ workgen sources into static library.
+add_library(workgen_cxx
+    STATIC
```

Comment:
format np: you probably want `workgen_cxx` above `STATIC` like you did lines 75/76

Reply:
Restyled.